### DISCUSSION TITLE
Possible to control the recirculation pump ?

### DISCUSSION THREAD

**Discussion by b0naf1de** [nikshriv/hass_navien_water_heater#24] [General]
Hello. I have a NPE-240A2 which includes the recirculation pump. 

I see references to the recirc pump in switch.py, but I am not sure what devices or entities should appear in home assistant to control the recirc pump. My instance provisioned a single device (CH1) with Power On/Off control and 8 entities. 

Can this integration control the recirc pump?
If so, how should it appear in home assistant?

**Comment by b0naf1de** [comment]
Answering my own question. If the Navien unit recirculation is configured for Hot Button (manual) mode, the integration will create an entity that acts like a hot buttton to trigger the pump. The Hot Button mode can be enabled from the normal menu's recirculation pump settings. However, when the NPE-240A2 is in this mode, the other modes are disabled (weekly schedule, intelligent). So, you have to create an automation to trigger the hot button if you want to have the pump run at certain time intervals or based on other events.

**Reply by ricracrum** [reply]
Thank you for following up on your original post. I'm just looking into this and have not installed this in HA yet. I also have an NPE-240A2 (with PCB). Reading forums, a complaint I see is that the Navilink app does not simultaneously support the Weekly Schedule AND Hot Button (to fire things up when the schedule has recirculation off. The only way to get the hot button to show in the app is to set the unit to Hot Button mode, which disables Weekly Schedule mode). In other words, pick one. However, your post here seems to indicate that, with the HA integration, one could put the unit in Hot Button mode and then build a weekly schedule automation in HA, and have that hot button available both from HA and the APP. Have I got that right? Thanks.

**Comment by philgrocks** [comment]
I didn’t try controlling everything with HA.

At the time I tried the Navilink integration I found it was slowing down startup. So I removed the integration.   Later I saw messages that the slow downs were improved.  

Perhaps this winter I will try again.

—
FWIW, I found that the Navien water heaters I have are super efficient.   So I have been happy with just scheduled operation.  Leaving the recirc on during our waking hours and off when we sleep.

Best

**Reply by ricracrum** [reply]
Thanks for your reply. I'm going to give it a shot and will report back. In my case, I have 2 units in a second home, and want to be able to stop recirculation (along with a variety of other house systems) as part of an "away" automation.

**Comment by tboyko** [comment]
@b0naf1de I believe #50 addresses your desire.

I set up my Navien / NaviLink for weekly schedule functionality (and not hot button). 

I wanted to have the recirc enabled during awake hours, but only when we're home. Initially I had home assistant power on and off the Navien based on geo, but this meant no hot water for appliances nor for guests remaining at the house.

The way Navien implemented it is counterintuitive to me: When the weekly function is enabled, the recirc is dictated by the schedule. When it's disabled, the recirc functionality is _enabled_ 24/7.

I keep the Navien power on, have my recirc schedule set to off for the entire week (a single OFF entry at midnight on Monday). I have the following rules in home assistant to turn the week function:

1. On once everyone has _left_ the house
2. Off when anyone is _arriving_ home
3. On in the late evening
4. Off in the morning assuming someone is home

It's been working well so far.

**Comment by tjstheiii** [comment]
my whatever they put under my sink, makes crazy ass noises. 99% of the time
if I want hot water I just turn on the cold faucet. 75% of the time it is
full of hot water and the hot water side is ice cold (and turns the stupid
thing on... to waste my money) I will say that this was the single worst
purchase of my life. I could have bought 5 water heaters in line and had
unlimited water for a lot less. Installers charged 4000 for 2men X 4h. They
did not ask me if I wanted this stupid noise maker on my bathroom sink.
Honestly, run before getting a unlimited water supply after a minute and
hot water coming out of your cold tap when you don't want it. If the
installation was -500 I would still decline without any thought. Worst
decision of my life. Their fucking "keep line hot and dump cold water (105
deg) into the cold line is stupid. If you enable it with the navilink and
150 dollar little device that does nothing except take up space you should
know if you turn that feature on for good... it voids your warranty. And
for whatever reason after you stop it it continues to ruin your life.
The damn thing like makes the weirdest sounds as it turns to let all the
mostly warm water into the cold return. I do not want to change your plan,
but just know what you're getting into and in my case I could have bought 6
giant water heaters for what I was charged. or two for the price of navi. I
can not see any advantage to get this unless you plan to live in your home
for 50 years and you do the maintenance every year and IDK. Think this
through.
